lib/tasks/load_matches.rake

```diff
@@ -2,51 +2,91 @@
 
 require Rails.root.join("lib/clients/apifutebol/api_client.rb")
 
-namespace :import do
-  desc "Import championship matches"
-  task matches: :environment do
-    matches = Client.new.matches
-
-    puts "Importing league '#{matches[0]['league_name']}'(id=#{matches[0]['league_id']})..."
-    list = matches[0]["stage"][0]["matches"]
-    puts "Matches: #{list.size}"
-    list.each do |m|
-      puts "Importing #{m['teams']['home']['name']} x #{m['teams']['away']['name']} at #{m['date']} #{m['time']}..."
-      match = Match.find_by(reference: m["id"])
-      unless match.nil?
-        puts "Match already imported, skipping..."
-        next
-      end
+namespace :import do # rubocop:disable all
+  desc "Import entire league data"
+  task league: :environment do
+    league = Client.new.league
 
-      team_home = find_or_create(m["teams"]["home"])
-      team_away = find_or_create(m["teams"]["away"])
+    puts "Importing league '#{league.name}'(id=#{league.id})..."
 
-      date = to_date(m["date"], m["time"])
-      Match.create(
-        date:,
-        team_home:,
-        team_away:,
-        status: m["status"],
-        home_goals: m["goals"]["home_ft_goals"],
-        away_goals: m["goals"]["away_ft_goals"],
-        result: m["winner"],
-        reference: m["id"]
+    puts "Teams: #{league.teams.size}"
+    teams = import_teams(league.teams)
+
+    puts "Matches: #{league.matches.size}"
+    import_matches(league.matches, teams)
+  end
+
+  desc "Import newly played matches"
+  task matches: :environment do
+    cli = Client.new
+    # Get matches that have been played but don't have results updated
+    matches = Match.pending.played
+    matches.each do |match|
+      puts "Importing #{match.team_home.name} x #{match.team_away.name} at #{match.date}"
+      m = cli.match(match.reference)
+      match.update(
+        status: m.status,
+        home_goals: m.home_goals,
+        away_goals: m.away_goals,
+        result: m.result
       )
+      if match.changed?
+        puts "Updating #{match.team_home.name} x #{match.team_away.name}..."
+        match.save
```

Comment:
redundant: `update` already saved the record

Reply:
Ahhh... so, I would need to use the attribution if I wanted to check for changes and update. Got it.

Reply:
Solved at #14

Reply:
If you want to have the code respond to the changes happening, then you can check if the result of update is `true` or `false`.  If the record fails to update, and the value is false, the object will have an array of `errors` that you can use to display or log the reasons for the save failure.

Example:

```rb
match = Match.find(id)
params = { your: :params, fill: "this hash"}
if match.update(**params)
  puts "Updated #{match.team_home.name} x #{match.team_away.name}"
else
  puts match.errors.full_messages.join("\n")
end
```

Or, if an update failure would be exceptional, you would use `update!` which throws an error on a failed DB write:

```rb
begin
  match = Match.find(id)
  params = { your: :params, fill: "this hash"}
  match.update!(**params)
rescue StandardError => e
   puts "Failed to update match: #{e.message}"
end
```